app/src/sailfish-qml/qml/harbour-musikloud2.qml

```diff
@@ -14,7 +15,52 @@ ApplicationWindow
     AudioPlayer {
         id: player
 
-        onStatusChanged: if (status == AudioPlayer.Failed) infoBanner.showError(errorString);
+        property SoundCloudTrackModel searchResults
+
+        onStatusChanged: {
+            if (status == AudioPlayer.Failed) infoBanner.showError(errorString);
+        }
+
+        onCurrentIndexChanged: {
+            // Fetch more tracks in current tracks to play are all played
+            var model = player.getSearchResults();
+            if (model && (player.currentIndex + 1) >= model.rowCount()) {
+                console.log("Fetching more tracks")
+                model.fetchMore()
+            }
+        }
+
+        function setTracksToPlay(trackModel) {
```

Comment:
Add argument fromIndex and use it below
for (var i = fromIndex ...)

and then remove the clearQueue or only do it if fromIndex is 0

then in _addFetchedResults you can set the from index as currentIndex + 1 and you don't have to manually remember the index.